### ISSUE TITLE
[BUG] filter_show_dates_ending

### ISSUE THREAD

**Issue by colinhatfield** [pgorod/power-todoist-card#4]
**Before submitting a bug**
- [ Yes] I updated to the latest card version available
- [ Yes] I cleared the cache of my browser

**Describe the bug**
Filtering by Dates - filter_show_dates_ending and filter_show_dates_ending do not seem to have any effect

**To Reproduce**
1. Go to any dashboard
2. Click on add card
3. Select custom: powertodoist card
4. Select the correct entity
5. Click Show Code Editor
6. Add filter_show_dates_ending: '0'
7. Add show_dates: true
8. Click Save

**Expected behavior**
I expected the card to show items due Today, or already overdue

**Versions:**
 - Card: v0.10-beta
 - HomeAssistant: 2023.10.5
 - Browser: Google Chrome 118.0.5993.118

**Comment by colinhatfield** [comment]
<img width="373" alt="Card" src="https://github.com/pgorod/power-todoist-card/assets/48314728/dc84ef27-6241-4f3a-9552-6d78dfd443ac">

**Comment by pgorod** [comment]
Hi. This seems to work for me so I need to ask for a bit more focused in information.

- Please focus on **a single item** that is not behaving as expected.
- tell me at what day/time you made your experiment ("now" is a most relevant concept here)
- clearly say what you expected, and what you got (things like "I expected it to show, it didn't" or the opposite)
- give me the full information on the item as it appears in JSON format in `HASS / Developer Tools / States / <name of your todoist sensor entity>`

This is what the screen looks like, after typing the sensor name in the filter box:
![image](https://github.com/pgorod/power-todoist-card/assets/15945027/76138ffb-7e06-4a99-b4ba-7e2735f4ed60)

An individual item's information looks like this:
```
- added_at: '2023-06-18T14:31:46.339183Z'
  added_by_uid: '43790529'
  assigned_by_uid: null
  checked: false
  child_order: 1
  collapsed: false
  completed_at: null
  content: Try a brand new Kanban in HASS! 💪
  description: ''
  due: null
  duration: null
  id: '6975761078'
  is_deleted: false
  labels:
    - Blue
  note_count: 0
  parent_id: null
  priority: 1
  project_id: '2314757944'
  responsible_uid: null
  section_id: '126441679'
  sync_id: null
  updated_at: '1970-01-01T00:00:00Z'
  user_id: '43790529'
```

**Comment by colinhatfield** [comment]
Thank-you for your quick response.

This is the item I'm looking at, I have just checked now (31/10/23 - 12:04) as the due date is 25/11/23 I am not expecting to see it displayed but it is displayed.
<img width="176" alt="Check" src="https://github.com/pgorod/power-todoist-card/assets/48314728/79fbf94e-3738-4eeb-9a52-48f0f9abbe37">
`- added_at: '2023-10-30T08:45:24.174761Z'
  added_by_uid: '46871525'
  assigned_by_uid: null
  checked: false
  child_order: 3
  collapsed: false
  completed_at: null
  content: Check Subscribe & Save
  description: ''
  due:
    date: '2023-11-25'
    is_recurring: true
    lang: en
    string: every 25th
    timezone: null
  duration: null
  id: '7362314182'
  is_deleted: false
  labels: []
  note_count: 0
  parent_id: null
  priority: 1
  project_id: '2322783911'
  responsible_uid: null
  section_id: '138223094'
  sync_id: null
  updated_at: '2023-10-30T08:47:01Z'
  user_id: '46871525'`

**Comment by pgorod** [comment]
Fixed in the latest v0.11-beta ! 

Thanks for your help, I had actually broken the entire date filtering code in a last-minute refactoring, and I hadn't noticed it yet...
